ngraph/frontend/paddlepaddle/src/frontend.cpp

```diff
@@ -135,20 +134,20 @@ std::shared_ptr<Function>
                 // inputs and outputs are stored in the model already
                 continue;
             } else {
-                const auto& node = pdpd::make_ng_node(nodes_dict, op_place, CREATORS_MAP);
-                if (node) {
-                    // set layer name by the name of first output var
-                    const auto& first_output_var = op_place->getOutputPorts().begin()->second[0]->getTargetTensorPDPD()->getDesc();
-                    node->set_friendly_name(first_output_var->name());
-
-                    std::cerr << "Named with " << node->get_friendly_name() << "\n";
-                    for (const auto& name_to_ports : op_place->getOutputPorts()) {
-                        int idx = 0;
-                        for (const auto& port : name_to_ports.second) {
-                            const auto& var = port->getTargetTensorPDPD()->getDesc();
-                            nodes_dict[var->name()] = node->output(idx);
-                            idx++;
-                        }
+                const auto& named_outputs = pdpd::make_ng_node(nodes_dict, op_place, CREATORS_MAP);
+                // set layer name by the name of first output var
+                const auto& first_output_var = op_place->getOutputPorts().begin()->second.at(0)->getTargetTensorPDPD()->getDesc();
+                const auto& node = named_outputs.begin()->second[0].get_node_shared_ptr();
```

Comment:
`named_outputs` can be empty, please add if statement here.

Reply:
fixed